var/spack/repos/builtin/packages/intel/package.py

```diff
@@ -0,0 +1,113 @@
+            mkl_components = "intel-mkl__x86_64;intel-mkl-ps__x86_64;intel-mkl-ps-jp__x86_64;intel-mkl-common__noarch;intel-mkl-ps-common__noarch;intel-mkl-ps-common-jp__noarch;intel-mkl-devel__x86_64;intel-mkl-ps-mic-devel__x86_64;intel-mkl-ps-f95-devel__x86_64;intel-mkl-gnu-devel__x86_64;intel-mkl-ps-gnu-devel__x86_64;intel-mkl-ps-pgi-devel__x86_64;intel-mkl-ps-cluster-devel__x86_64;intel-mkl-ps-cluster-common__noarch;intel-mkl-ps-f95-common__noarch;intel-mkl-ps-cluster__x86_64;intel-mkl-gnu__x86_64;intel-mkl-ps-gnu__x86_64;intel-mkl-ps-pgi__x86_64;intel-mkl-ps-mic__x86_64;intel-mkl-ps-mic-jp__x86_64;intel-mkl-doc__noarch;intel-mkl-ps-doc__noarch;intel-mkl-ps-doc-jp__noarch;intel-mkl-ps-ss-tbb__x86_64;intel-mkl-ps-ss-tbb-devel__x86_64;intel-mkl-ps-tbb-mic__x86_64;intel-mkl-ps-tbb-mic-devel__x86_64"
+            daal_components = "intel-daal__x86_64;intel-daal-common__noarch;intel-daal-ps-common-jp__noarch;intel-daal-doc__noarch;intel-daal-ps-doc-jp__noarch"
+            ipp_components = "intel-ipp-l-common__noarch;intel-ipp-l-ps-common__noarch;intel-ipp-l-st__x86_64;intel-ipp-l-mt__x86_64;intel-ipp-l-st-devel__x86_64;intel-ipp-l-ps-st-devel__x86_64;intel-ipp-l-mt-devel__x86_64;intel-ipp-l-doc__noarch;intel-ipp-l-ps-doc-jp__noarch"
+            tools_components = "intel-gdb-gt__x86_64;intel-gdb-gt-libelfdwarf__x86_64;intel-gdb-gt-devel__x86_64;intel-gdb-gt-common__noarch;intel-gdb-gt-doc__noarch;intel-gdb-ps-cdt__x86_64;intel-gdb-ps-cdt-source__x86_64;intel-gdb-ps-mic__x86_64;intel-gdb-ps-mpm__x86_64;intel-gdb-ps-doc__noarch;intel-gdb-ps-doc-jp__noarch;intel-gdb__x86_64;intel-gdb-common__noarch;intel-gdb-doc__noarch;intel-gdb-doc-jp__noarch;intel-gdb-gt-src__noarch;intel-gdb-source__noarch;intel-compxe__noarch;intel-gdb-ps-common__noarch;intel-vtune-amplifier-xe-2016-cli__i486;intel-vtune-amplifier-xe-2016-cli__x86_64;intel-vtune-amplifier-xe-2016-common__noarch;intel-vtune-amplifier-xe-2016-cli-common__noarch;intel-vtune-amplifier-xe-2016-collector-32linux__i486;intel-vtune-amplifier-xe-2016-collector-64linux__x86_64;intel-vtune-amplifier-xe-2016-doc__noarch;intel-vtune-amplifier-xe-2016-sep__noarch;intel-vtune-amplifier-xe-2016-gui-common__noarch;intel-vtune-amplifier-xe-2016-gui__x86_64;intel-vtune-amplifier-xe-2016-common-pset__noarch;intel-inspector-xe-2016-cli__i486;intel-inspector-xe-2016-cli__x86_64;intel-inspector-xe-2016-cli-common__noarch;intel-inspector-xe-2016-doc__noarch;intel-inspector-xe-2016-gui-common__noarch;intel-inspector-xe-2016-gui__x86_64;intel-inspector-xe-2016-cli-pset__noarch;intel-advisor-xe-2016-cli__i486;intel-advisor-xe-2016-cli__x86_64;intel-advisor-xe-2016-cli-common__noarch;intel-advisor-xe-2016-doc__noarch;intel-advisor-xe-2016-gui-common__noarch;intel-advisor-xe-2016-gui__i486;intel-advisor-xe-2016-gui__x86_64;intel-advisor-xe-2016-cli-pset__noarch"
```

Comment:
I doubt it, but haven't tried. I did find that within the Intel distribution, one can `grep Abbr pset/mediaconfig.xml` to see what all of the potential components are. If we really wanted to, we can use this list to gather a list of all possible components and then do our own filtering to choose which ones we want. This will be much cleaner than manually specifying them all. I will work on this.